extensions/sql/asset-index-sql/src/test/java/org/eclipse/dataspaceconnector/sql/assetindex/schema/postgres/PostgresAssetIndexTest.java

```diff
@@ -0,0 +1,205 @@
+    private static final String JDBC_URL_PREFIX = "jdbc:postgresql://localhost:5432/";
+    private SqlAssetIndex sqlAssetIndex;
+    private BaseSqlDialectStatements sqlStatements;
+    private TransactionContext transactionContext;
+    private Connection connection;
+
+    @BeforeAll
+    static void prepare() {
+        // todo: reuse org.eclipse.dataspaceconnector.test.e2e.postgresql.PostgresqlLocalInstance??
+        try (var connection = DriverManager.getConnection(JDBC_URL_PREFIX + POSTGRES_USER, POSTGRES_USER, POSTGRES_PASSWORD)) {
+            connection.createStatement().execute(format("CREATE DATABASE %s;", POSTGRES_DATABASE));
+        } catch (SQLException e) {
+            // database could already exist
+        }
+    }
+
+
+    @BeforeEach
+    void setUp() throws SQLException, IOException {
+        transactionContext = new NoopTransactionContext();
+        DataSourceRegistry dataSourceRegistry = mock(DataSourceRegistry.class);
+
+
+        var ds = new PGSimpleDataSource();
+        ds.setServerNames(new String[]{ "localhost" });
+        ds.setPortNumbers(new int[]{ 5432 });
+        ds.setUser(POSTGRES_USER);
+        ds.setPassword(POSTGRES_PASSWORD);
+        ds.setDatabaseName(POSTGRES_DATABASE);
+
+        // do not actually close
+        connection = spy(ds.getConnection());
+        doNothing().when(connection).close();
+
+        var datasourceMock = mock(DataSource.class);
+        when(datasourceMock.getConnection()).thenReturn(connection);
+        when(dataSourceRegistry.resolve(DATASOURCE_NAME)).thenReturn(datasourceMock);
+
+        sqlStatements = new PostgresDialectStatements();
+        TypeManager manager = new TypeManager();
+
+        manager.registerTypes(PolicyRegistrationTypes.TYPES.toArray(Class<?>[]::new));
+        sqlAssetIndex = new SqlAssetIndex(dataSourceRegistry, DATASOURCE_NAME, transactionContext, new ObjectMapper(), sqlStatements);
+        var schema = Files.readString(Paths.get("./docs/schema.sql"));
+        try {
+            transactionContext.execute(() -> {
+                executeQuery(connection, schema);
+                return null;
+            });
+        } catch (Exception exc) {
+            fail(exc);
+        }
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+
+        transactionContext.execute(() -> {
+            var dialect = new PostgresDialectStatements();
+            executeQuery(connection, "DROP TABLE " + dialect.getAssetTable() + " CASCADE");
+            executeQuery(connection, "DROP TABLE " + dialect.getDataAddressTable() + " CASCADE");
+            executeQuery(connection, "DROP TABLE " + dialect.getAssetPropertyTable() + " CASCADE");
+        });
+        doCallRealMethod().when(connection).close();
+        connection.close();
+    }
+
+    @Test
+    void query_byAssetProperty() {
+        var allAssets = IntStream.range(0, 5).mapToObj(i -> {
+            var asset = createAssetBuilder("test-asset" + i)
```

Comment:
this function that maps an integer to an integer to an asset and insert it into the asset index could be extracted as it is used 3 times (not in the same way, but very similarly)